src/rest_vol_link.c

```diff
@@ -73,8 +73,11 @@ const char *link_collection_keys2[] = { "collection", (const char *) 0 };
  *              July, 2017
  */
 herr_t
-RV_link_create(H5VL_link_create_type_t create_type, void *obj, const H5VL_loc_params_t *loc_params,
-    hid_t lcpl_id, hid_t lapl_id, hid_t dxpl_id, void **req, va_list arguments)
+RV_link_create(H5VL_link_create_args_t *args, void *obj, const H5VL_loc_params_t *loc_params,
+                  const H5VL_class_t *cls, hid_t lcpl_id, hid_t lapl_id, hid_t dxpl_id, void **req)
+
+/* (H5VL_link_create_t create_type, void *obj, const H5VL_loc_params_t *loc_params,
+    hid_t lcpl_id, hid_t lapl_id, hid_t dxpl_id, void **req, va_list arguments) */
```

Comment:
There's a few of these commented out old prototypes that we should just remove now.